ios/FluentUI/Navigation/BarButtonItemView.swift

```diff
@@ -0,0 +1,271 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+class BarButtonItemView: UIView {
+    let item: UIBarButtonItem
+    let isLeftItem: Bool
+
+    init(for item: UIBarButtonItem, isLeftItem: Bool) {
+        self.item = item
+        self.isLeftItem = isLeftItem
+        super.init(frame: .zero)
+
+        container.addArrangedSubview(button)
+        container.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(container)
+
+        container.addSubview(badgeView)
```

Comment:
This is exactly the same behavior in the case of badge coming on side tab bar items but yes I do think that press effect should come on the whole thing. Also, designers are OK with the current thing but yes I can create a VSO item and we can make this enhancement in the next PR.

Reply:
It's not exactly the same behavior as the SideTabBar because the items of the SideTabBar don't animate the icon (or the badge) while it's being pressed. This is the intrinsic behavior of the UIButton, and the SideTabBar uses UIViews with tap gesture recognizers.

Below a demo of what I mean. The buttons on the navigation controller are dimming the contents (icon and "Dismiss" text) while the icon in the tab bar doesn't change until the tap event is completed and the state of the view changes to selected.


https://user-images.githubusercontent.com/68076145/133859697-37e25598-9005-4760-b416-98aec9678baf.mov